donkeycar/templates/complete.py

```diff
@@ -585,21 +585,13 @@ def run(self, mode, recording):
         cfg.AUTO_CREATE_NEW_TUB else cfg.DATA_PATH
     tub_writer = TubWriter(tub_path, inputs=inputs, types=types, metadata=meta)
     V.add(tub_writer, inputs=inputs, outputs=["tub/num_records"], run_condition='recording')
-    # Telemetry (we add the same metrics added to TubHandler
-    if cfg.HAVE_TELEMETRY:
-        from donkeycar.parts.telemetry import Telemetry
-
-        # publish all simple types
-        inputs_to_publish = []
-        types_to_publish = []
-        for ind in range(0, len(types)):
-            if types[ind] in ['float', 'str', 'int']:
-                inputs_to_publish.append(inputs[ind])
-                types_to_publish.append(types[ind])
-
-        tel = Telemetry(cfg, default_inputs=inputs_to_publish, default_types=types_to_publish)
-        V.add(tel, inputs=inputs_to_publish, outputs=["tub/queue_size"], threaded=False)
 
+    # Telemetry (we add the same metrics added to the TubHandler
+    if cfg.HAVE_MQTT_TELEMETRY:
+        from donkeycar.parts.telemetry import MqttTelemetry
+        published_inputs, published_types = MqttTelemetry.filter_supported_metrics(inputs, types)
```

Comment:
Why don't you do this inside the constructor for `MqttTelemetry` itself, instead of doing this in `complete.py`.

Reply:
Because `published_inputs` are passed to `V.add(tel, inputs=published_inputs`.... This why only relevant arguments are passed to `run`...

Reply:
I mean you can pass in the input metrics as is to the constructor of `MqttTelemetry` and filter it in the __init__ block. 

The `run` method can  remain the same.

Reply:
Sorry about that, I looked at the code again and now I understand what you are referring to. This looks okay to me.